### ISSUE TITLE
RandomCrop bugs

### ISSUE THREAD

**Issue by martin-gorner** [keras-team/keras-cv#860]
Repro Colab [here](https://colab.research.google.com/drive/1fi7CRtVzDH9xLTk1g2yae671HONpuppN#scrollTo=_uBFDKhkbriw)

Input image size: 360*512px

Problem #1: cropping to 128*513 crashes with a div by 0
Problem #2: cropping to 128*514 no longer produces random results (there are still many possible random positions along the Y axis)

**Comment by bhack** [comment]
Yes for 1 it is an edge case of `get_random_transformation` in Keras
https://github.com/keras-team/keras/blob/v2.10.0/keras/layers/preprocessing/image_preprocessing.py#L581-L582
`w_start = rands[1] % (w_diff + 1)`

As your image in the colab cell is 
```python 
tf.Tensor([372 512   3], shape=(3,), dtype=int32)
```
With
```python 
random_crop = tf.keras.layers.RandomCrop(height=128, width=513)
```

You are going to have a `w_diff` of `-1` and so you crash with `% 0`

**Comment by bhack** [comment]
For 2 I I don't understand well the logic of using  `dtype.max`
https://github.com/keras-team/keras/blob/v2.10.0/keras/layers/preprocessing/image_preprocessing.py#L580

```python
rands = self._random_generator.random_uniform([2], 0, dtype.max, dtype)
```

**Comment by LukeWood** [comment]
Thank you Martin for the bug report!

**Comment by sachinprasadhs** [comment]
Thanks for reporting the issue! We have consolidated the development of KerasCV into the new **KerasHub** package, which supports image, text, and multi-modal models. Please read [the announcement](https://github.com/keras-team/keras-hub/issues/1831). `KerasHub` will support all the core functionality of `KerasCV`.

`KerasHub` can be installed with `!pip install -U keras-hub`. Documentation and guides are available at [keras.io/keras_hub](https://keras.io/keras_hub/).

With our focus shifted to KerasHub, we are not planning any further development or releases in KerasCV. If you encounter a KerasCV feature that is missing from KerasHub, or would like to propose an addition to the library, please file [an issue with KerasHub](https://github.com/keras-team/keras-hub/issues).
